syntax = "proto3";

import "google/protobuf/empty.proto";
package cs739;

service ExchangeInfo {
  rpc ExchangeInteger(Integer) returns(Integer) {}
  rpc ExchangeDouble(Double) returns(Double) {}
  rpc ExchangeString(String) returns(String) {}
  rpc ExchangeComplex(Complex) returns(Complex) {}
  rpc RoundTrip(google.protobuf.Empty) returns(Time) {}
}

message Integer {
  int32 data = 1;
}

message Double { 
  double data = 1;
}

message String {
  string data = 1;
}

message Complex {
  int32 i = 1;
  double d = 2;
  string s = 3;
}

message Time {
  int64 sec = 1;
  int64 nsec = 2;
}
